### PR TITLE
Lint: enable errcheck linter

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <long0dai@foxmail.com>

**What type of PR is this?**

> /kind cleanup

**What this PR does / why we need it**:

Enable errcheck linter.

### CODE REVIEW

File: cloud/pkg/cloudhub/channelq/channelq.go [hunk 1]
```diff
@@ -84,7 +84,9 @@ func (q *ChannelMessageQueue) addListMessageToQueue(nodeID string, msg *beehiveM
 
 	messageKey, _ := getListMsgKey(msg)
 
-	nodeListStore.Add(msg)
+	if err := nodeListStore.Add(msg); err != nil {
+		klog.Errorf("failed to add msg: %s", err)
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/channelq/channelq.go:88]: klog.Errorf("Failed to add msg, err: %v", err)

Comment (fisherxu) [reply cloud/pkg/cloudhub/channelq/channelq.go:88]: Should return here.

File: cloud/pkg/cloudhub/servers/httpserver/signcerts.go [hunk 2]
```diff
@@ -80,8 +80,8 @@ func GenerateToken() error {
 	caHashToken := strings.Join([]string{caHash, tokenString}, ".")
 	// save caHashAndToken to secret
 	err = CreateTokenSecret([]byte(caHashToken))
-	if err != nil {
-		return fmt.Errorf("Failed to create tokenSecret, err: %v", err)
+	if err := CreateTokenSecret([]byte(caHashToken)); err != nil {
+		klog.Fatalf("Failed to create the ca token  for edgecore register, err: %v", err)
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/signcerts.go:84]: The point call the CreateTokenSecret wil fatel, return here is enough.

File: cloud/pkg/cloudhub/servers/udsserver/server.go [hunk 3]
```diff
@@ -42,7 +42,10 @@ func StartServer(address string) {
 	})
 
 	klog.Info("start unix domain socket server")
-	uds.StartServer()
+	if err := uds.StartServer(); err != nil {
+		klog.Errorf("Failed to start server: %v", err)
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/udsserver/server.go:46]: klog.Errorf("Failed to start uds server: %v", err)

File: cloud/pkg/devicecontroller/controller/downstream.go [hunk 4]
```diff
@@ -657,7 +657,10 @@ func (dc *DownstreamController) deleteFromConfigMap(device *v1alpha1.Device) {
 		// no device is bound to the configMap, then remove the configMap directly.
 		if nodeConfigMap.Data[DeviceProfileJSON] == "{}" {
 			deleteOptions := &metav1.DeleteOptions{}
-			dc.kubeClient.CoreV1().ConfigMaps(device.Namespace).Delete(nodeConfigMap.Name, deleteOptions)
+			if err := dc.kubeClient.CoreV1().ConfigMaps(device.Namespace).Delete(nodeConfigMap.Name, deleteOptions); err != nil {
+				klog.Errorf("Failed to delete config map %v in namespace %v", nodeConfigMap, device.Namespace)
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/controller/downstream.go:661]: klog.Errorf("Failed to delete config map %s in namespace %s", nodeConfigMap.name, device.Namespace)

File: cloud/pkg/devicecontroller/devicecontroller.go [hunk 5]
```diff
@@ -57,9 +57,15 @@ func (dc *DeviceController) Start() {
 		os.Exit(1)
 	}
 
-	downstream.Start()
+	if err := downstream.Start(); err != nil {
+		klog.Errorf("start downstream failed with error: %s", err)
+		os.Exit(1)
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/devicecontroller.go:62]: fatel

File: cloud/pkg/devicecontroller/devicecontroller.go [hunk 6]
```diff
@@ -57,9 +57,15 @@ func (dc *DeviceController) Start() {
...
+	}
 	// wait for downstream controller to start and load deviceModels and devices
 	// TODO think about sync
 	time.Sleep(1 * time.Second)
-	upstream.Start()
+	if err := upstream.Start(); err != nil {
+		klog.Errorf("start upstream failed with error: %s", err)
+		os.Exit(1)
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/devicecontroller.go:69]: fatel?

File: cloud/pkg/edgecontroller/edgecontroller.go [hunk 7]
```diff
@@ -51,12 +51,19 @@ func (ec *EdgeController) Start() {
 		klog.Errorf("new upstream controller failed with error: %s", err)
 		os.Exit(1)
 	}
-	upstream.Start()
+
+	if err := upstream.Start(); err != nil {
+		klog.Errorf("start upstream failed with error: %s", err)
+		os.Exit(1)
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/edgecontroller.go:57]: ditto

File: cloud/pkg/edgecontroller/edgecontroller.go [hunk 8]
```diff
@@ -51,12 +51,19 @@ func (ec *EdgeController) Start() {
...
 
 	downstream, err := controller.NewDownstreamController()
 	if err != nil {
 		klog.Warningf("new downstream controller failed with error: %s", err)
 		os.Exit(1)
 	}
-	downstream.Start()
+	if err := downstream.Start(); err != nil {
+		klog.Errorf("start downstream failed with error: %s", err)
+		os.Exit(1)
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/edgecontroller.go:67]: ditto

File: edge/pkg/devicetwin/dtmanager/device.go [hunk 9]
```diff
@@ -107,16 +107,18 @@ func dealDeviceStateUpdate(context *dtcontext.DTContext, resource string, msg in
 
 	}
 	topic := dtcommon.DeviceETPrefix + device.ID + dtcommon.DeviceETStateUpdateSuffix + "/result"
-	context.Send(device.ID,
-		dtcommon.SendToEdge,
-		dtcommon.CommModule,
-		context.BuildModelMessage(modules.BusGroup, "", topic, "publish", payload))
+	if err = context.Send(device.ID, dtcommon.SendToEdge, dtcommon.CommModule,
+		context.BuildModelMessage(modules.BusGroup, "",
+			topic, "publish", payload)); err != nil {
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/device.go:112]: I think maybe no need delete the `line-break` here.

File: edge/pkg/devicetwin/dtmanager/device.go [hunk 10]
```diff
@@ -107,16 +107,18 @@ func dealDeviceStateUpdate(context *dtcontext.DTContext, resource string, msg in
...
 	msgResource := "device/" + device.ID + "/state"
-	context.Send(deviceID,
-		dtcommon.SendToCloud,
-		dtcommon.CommModule,
-		context.BuildModelMessage("resource", "", msgResource, "update", string(payload)))
+	if err = context.Send(deviceID, dtcommon.SendToCloud, dtcommon.CommModule,
+		context.BuildModelMessage("resource", "",
+			msgResource, "update", string(payload))); err != nil {
+		return nil, err
+	}
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/device.go:121]: ditto

File: edge/pkg/devicetwin/dtmanager/device.go [hunk 11]
```diff
@@ -176,8 +180,11 @@ func DeviceUpdated(context *dtcontext.DTContext, deviceID string, attributes map
 				klog.Errorf("Build device attribute update failed: %v", err)
 			}
 			topic := dtcommon.DeviceETPrefix + deviceID + dtcommon.DeviceETUpdatedSuffix
-			context.Send(deviceID, dtcommon.SendToEdge, dtcommon.CommModule,
-				context.BuildModelMessage(modules.BusGroup, "", topic, "publish", payload))
+			if err := context.Send(deviceID, dtcommon.SendToEdge, dtcommon.CommModule,
+				context.BuildModelMessage(modules.BusGroup, "",
+					topic, "publish", payload)); err != nil {
+				return nil, err
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/device.go:186]: ditto

File: edge/pkg/devicetwin/dtmanager/membership.go [hunk 12]
```diff
@@ -182,8 +184,12 @@ func Added(context *dtcontext.DTContext, toAdd []dttype.Device, baseMessage dtty
 				klog.Errorf("Add device %s failed, has existed", device.ID)
 				continue
 			}
-			DeviceUpdated(context, device.ID, device.Attributes, baseMessage, dealType)
-			DealDeviceTwin(context, device.ID, baseMessage.EventID, device.Twin, dealType)
+			if _, err := DeviceUpdated(context, device.ID, device.Attributes, baseMessage, dealType); err != nil {
+
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership.go:188]: Need handling errors here.

File: edge/pkg/devicetwin/dtmanager/membership.go [hunk 13]
```diff
@@ -182,8 +184,12 @@ func Added(context *dtcontext.DTContext, toAdd []dttype.Device, baseMessage dtty
...
+			}
+			if err := DealDeviceTwin(context, device.ID, baseMessage.EventID, device.Twin, dealType); err != nil {
+
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership.go:191]: ditto

File: edge/pkg/devicetwin/dtmanager/membership.go [hunk 14]
```diff
@@ -225,12 +231,16 @@ func Added(context *dtcontext.DTContext, toAdd []dttype.Device, baseMessage dtty
 		}
 		if device.Twin != nil {
 			klog.Infof("Add device twin during first adding device %s", device.ID)
-			DealDeviceTwin(context, device.ID, baseMessage.EventID, device.Twin, dealType)
+			if err := DealDeviceTwin(context, device.ID, baseMessage.EventID, device.Twin, dealType); err != nil {
+
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership.go:235]: ditto

File: edge/pkg/devicetwin/dtmanager/membership.go [hunk 15]
```diff
@@ -225,12 +231,16 @@ func Added(context *dtcontext.DTContext, toAdd []dttype.Device, baseMessage dtty
...
+			}
 		}
 
 		if device.Attributes != nil {
 			klog.Infof("Add device attr during first adding device %s", device.ID)
-			DeviceUpdated(context, device.ID, device.Attributes, baseMessage, dealType)
+			if _, err := DeviceUpdated(context, device.ID, device.Attributes, baseMessage, dealType); err != nil {
+
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership.go:242]: ditto

File: edge/pkg/devicetwin/dtmanager/membership.go [hunk 16]
```diff
@@ -241,10 +251,10 @@ func Added(context *dtcontext.DTContext, toAdd []dttype.Device, baseMessage dtty
 		if err != nil {
 
 		} else {
-			context.Send("",
-				dtcommon.SendToEdge,
-				dtcommon.CommModule,
-				context.BuildModelMessage(modules.BusGroup, "", topic, "publish", result))
+			if err := context.Send("", dtcommon.SendToEdge, dtcommon.CommModule,
+				context.BuildModelMessage(modules.BusGroup, "", topic, "publish", result)); err != nil {
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership.go:255]: no need to delete the `line-break`

File: edge/pkg/devicetwin/dtmanager/membership.go [hunk 17]
```diff
@@ -295,10 +305,10 @@ func Removed(context *dtcontext.DTContext, toRemove []dttype.Device, baseMessage
...
+				klog.Errorf("Send fail;failed: %v", err)
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership.go:310]: ditto

File: edge/pkg/devicetwin/dtmanager/membership.go [hunk 18]
```diff
@@ -344,10 +354,10 @@ func DealGetMembership(context *dtcontext.DTContext, payload []byte) error {
 	topic := dtcommon.MemETPrefix + context.NodeName + dtcommon.MemETGetResultSuffix
 	klog.Infof("Deal getting membership successful and send the result")
 
-	context.Send("",
-		dtcommon.SendToEdge,
-		dtcommon.CommModule,
-		context.BuildModelMessage(modules.BusGroup, "", topic, "publish", result))
+	if err := context.Send("", dtcommon.SendToEdge, dtcommon.CommModule,
+		context.BuildModelMessage(modules.BusGroup, "", topic, "publish", result)); err != nil {
+		return err
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership.go:359]: ditto

File: edge/pkg/devicetwin/dttype/types_helper.go [hunk 19]
```diff
@@ -71,25 +71,33 @@ func DeviceTwinToMsgTwin(deviceTwins []dtclient.DeviceTwin) map[string]*MsgTwin
 		if expected != "" {
 			expectedValue := &TwinValue{Value: &expected}
 			if twin.ExpectedMeta != "" {
-				json.Unmarshal([]byte(twin.ExpectedMeta), &expectedMeta)
+				if err := json.Unmarshal([]byte(twin.ExpectedMeta), &expectedMeta); err != nil {
+
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dttype/types_helper.go:75]: Handling the err

File: edge/pkg/devicetwin/dttype/types_helper.go [hunk 20]
```diff
@@ -71,25 +71,33 @@ func DeviceTwinToMsgTwin(deviceTwins []dtclient.DeviceTwin) map[string]*MsgTwin
...
 				expectedValue.Metadata = &expectedMeta
 			}
 			msgTwin.Expected = expectedValue
 		}
 		if actual != "" {
 			actualValue := &TwinValue{Value: &actual}
 			if twin.ActualMeta != "" {
-				json.Unmarshal([]byte(twin.ActualMeta), &actualMeta)
+				if err := json.Unmarshal([]byte(twin.ActualMeta), &actualMeta); err != nil {
+
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dttype/types_helper.go:85]: ditto

File: edge/pkg/devicetwin/dttype/types_helper.go [hunk 21]
```diff
@@ -71,25 +71,33 @@ func DeviceTwinToMsgTwin(deviceTwins []dtclient.DeviceTwin) map[string]*MsgTwin
...
-			json.Unmarshal([]byte(twin.ExpectedVersion), &expectedVersion)
+			if err := json.Unmarshal([]byte(twin.ExpectedVersion), &expectedVersion); err != nil {
+
+			}
 			msgTwin.ExpectedVersion = &expectedVersion
 		}
 		if twin.ActualVersion != "" {
-			json.Unmarshal([]byte(twin.ActualVersion), &actualVersion)
+			if err := json.Unmarshal([]byte(twin.ActualVersion), &actualVersion); err != nil {
+
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dttype/types_helper.go:99]: ditto

File: edgemesh/pkg/plugin/plugin.go [hunk 22]
```diff
@@ -1,6 +1,7 @@
 package plugin
 
 import (
+	"fmt"
```
Comment (fisherxu) [inline edgemesh/pkg/plugin/plugin.go:4]: empty line needed

File: cloud/pkg/cloudhub/channelq/channelq.go (hunk 1 above)
Comment (daixiang0) [reply cloud/pkg/cloudhub/channelq/channelq.go:88]: Error strings should not  be capitalized, see https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Comment (daixiang0) [reply cloud/pkg/cloudhub/channelq/channelq.go:88]: Add return.

File: edge/pkg/devicetwin/dttype/types_helper.go [hunk 23]
```diff
@@ -6,6 +6,7 @@ import (
 	"time"
 
 	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtclient"
+	"k8s.io/klog"
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dttype/types_helper.go:9]: Move to L7

File: edge/pkg/eventbus/mqtt/server.go [hunk 24]
```diff
@@ -143,5 +143,7 @@ func (m *Server) Publish(topic string, payload []byte) {
 		Payload: payload,
 		QOS:     packet.QOS(m.qos),
 	}
-	m.backend.Publish(client, msg, nil)
+	if err := m.backend.Publish(client, msg, nil); err != nil {
+
```
Comment (fisherxu) [inline edge/pkg/eventbus/mqtt/server.go:147]: Handling the err here.

File: edge/pkg/devicetwin/dttype/types_helper.go (hunk 23 above)
Comment (daixiang0) [reply edge/pkg/devicetwin/dttype/types_helper.go:9]: It is what gofmt does.

Comment (fisherxu) [reply edge/pkg/devicetwin/dttype/types_helper.go:9]: "time"

       "k8s.io/klog"

	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtclient"

**Comment by daixiang0** [comment]
Add errcheck make test fail.

**Comment by daixiang0** [comment]
Disable to check edge part, will need much time to fix them, let us enable errcheck linter first @fisherxu :)

**Comment by fisherxu** [comment]
It seems somethg wrong with keadm.

**Comment by daixiang0** [comment]
In fact, this [line](https://github.com/kubeedge/kubeedge/blob/master/keadm/cmd/keadm/app/keadm.go#L31) always fail, no `logtostderr` flag added.

**Comment by daixiang0** [comment]
ping  @fisherxu

**Comment by fisherxu** [comment]
lgtm, please squash the meaningless commits :)

File: keadm/cmd/keadm/app/keadm.go [hunk 25]
```diff
@@ -28,7 +28,6 @@ import (
 //Run executes the keadm command
 func Run() error {
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
-	pflag.Set("logtostderr", "true")
```
Comment (chendave) [inline keadm/cmd/keadm/app/keadm.go:31]: why remove this line?

File: edgemesh/pkg/proxy/proxy.go [hunk 26]
```diff
@@ -249,5 +257,7 @@ func Clean() {
 		}
 		nameserver = nameserver + item + "\n"
 	}
-	ioutil.WriteFile(hostResolv, []byte(nameserver), 0600)
+	if err := ioutil.WriteFile(hostResolv, []byte(nameserver), 0600); err != nil {
+		klog.Errorf("[EdgeMesh] write file %s err: %v", hostResolv, err)
```
Comment (chendave) [inline edgemesh/pkg/proxy/proxy.go:261]: how to "failed to write name server to file %s, error: %v"?

File: edgemesh/pkg/proxy/proxy.go [hunk 27]
```diff
@@ -209,7 +213,9 @@ func ensureResolvForHost() {
 		idx++
 	}
 
-	ioutil.WriteFile(hostResolv, []byte(nameserver), 0600)
+	if err := ioutil.WriteFile(hostResolv, []byte(nameserver), 0600); err != nil {
+
```
Comment (chendave) [inline edgemesh/pkg/proxy/proxy.go:217]: ?

File: edgemesh/pkg/proxy/proxy.go [hunk 28]
```diff
@@ -193,7 +195,9 @@ func ensureResolvForHost() {
 	if configured {
 		if dnsIdx != startIdx && dnsIdx > startIdx {
 			nameserver := sortNameserver(resolv, dnsIdx, startIdx)
-			ioutil.WriteFile(hostResolv, []byte(nameserver), 0600)
+			if err := ioutil.WriteFile(hostResolv, []byte(nameserver), 0600); err != nil {
```
Comment (chendave) [inline edgemesh/pkg/proxy/proxy.go:198]: better to log error here.

File: edgemesh/pkg/plugin/plugin.go [hunk 29]
```diff
@@ -41,7 +43,11 @@ func Install() {
...
 		Address: config.GlobalDefinition.Panel.Settings["address"],
 	}
-	control.Init(opts)
+	if err := control.Init(opts); err != nil {
+		_ = fmt.Errorf("failed to init control: %v", err)
+	}
 	// init archaius
-	archaius.Init()
+	if err := archaius.Init(); err != nil {
+		_ = fmt.Errorf("failed to init arahaius: %v", err)
```
Comment (chendave) [inline edgemesh/pkg/plugin/plugin.go:51]: s/arahaius/archaius

File: edgemesh/pkg/listener/listener.go [hunk 30]
```diff
@@ -256,7 +256,9 @@ func realServerAddress(conn *net.Conn) (string, int, error) {
 	}
 
 	port := int(addr.data[0])<<8 + int(addr.data[1])
-	syscall.SetNonblock(int(fd), true)
+	if err := syscall.SetNonblock(int(fd), true); err != nil {
```
Comment (chendave) [inline edgemesh/pkg/listener/listener.go:259]: better to log error.

File: edgemesh/pkg/dns/dns.go [hunk 31]
```diff
@@ -330,7 +333,9 @@ func getFromRealDNS(req []byte, from *net.UDPAddr) {
 
 		if n > 0 {
 			rsp = append(rsp, buf[:n]...)
-			dnsConn.WriteToUDP(rsp, from)
+			if _, err = dnsConn.WriteToUDP(rsp, from); err != nil {
+				continue
```
Comment (chendave) [inline edgemesh/pkg/dns/dns.go:337]: Does the err need to be logged here?

File: keadm/cmd/keadm/app/keadm.go (hunk 25 above)
Comment (daixiang0) [reply keadm/cmd/keadm/app/keadm.go:31]: It always returns err so it does not do impact :(

File: cloud/pkg/cloudhub/channelq/channelq.go [hunk 32]
```diff
@@ -140,7 +143,9 @@ func (q *ChannelMessageQueue) addMessageToQueue(nodeID string, msg *beehiveModel
 		}
 	}
 
-	nodeStore.Add(msg)
+	if err := nodeStore.Add(msg); err != nil {
+		return
```
Comment (chendave) [inline cloud/pkg/cloudhub/channelq/channelq.go:147]: prefer to logging the error message or else we don't know what's happened.

File: cloud/pkg/cloudhub/handler/messagehandler.go [hunk 33]
```diff
@@ -405,7 +405,9 @@ func (mh *MessageHandle) handleMessage(nodeQueue workqueue.RateLimitingInterface
 	if msgType == "listMessage" {
 		mh.send(hi, info, msg)
 		// delete successfully sent events from the queue/store
-		nodeStore.Delete(msg)
+		if err := nodeStore.Delete(msg); err != nil {
+			return
```
Comment (chendave) [inline cloud/pkg/cloudhub/handler/messagehandler.go:409]: same here.

File: cloud/pkg/cloudhub/handler/messagehandler.go [hunk 34]
```diff
@@ -469,7 +471,9 @@ func (mh *MessageHandle) saveSuccessPoint(msg *beehiveModel.Message, info *model
 		objectSyncName := synccontroller.BuildObjectSyncName(info.NodeID, resourceUID)
 
 		if msg.GetOperation() == beehiveModel.DeleteOperation {
-			nodeStore.Delete(msg)
+			if err := nodeStore.Delete(msg); err != nil {
+				return
```
Comment (chendave) [inline cloud/pkg/cloudhub/handler/messagehandler.go:475]: ditto.

File: cloud/pkg/cloudhub/servers/httpserver/server.go [hunk 35]
```diff
@@ -69,18 +69,22 @@ func StartHTTPServer() {
 // getCA returns the caCertDER
 func getCA(w http.ResponseWriter, r *http.Request) {
 	caCertDER := hubconfig.Config.Ca
-	w.Write(caCertDER)
+	_, _ = w.Write(caCertDER)
```
Comment (chendave) [inline cloud/pkg/cloudhub/servers/httpserver/server.go:72]: can you explain why this change is needed?

File: cloud/pkg/cloudhub/servers/httpserver/server.go [hunk 36]
```diff
@@ -69,18 +69,22 @@ func StartHTTPServer() {
...
 }
 
 //electionHandler returns the status whether the cloudcore is ready
 func electionHandler(w http.ResponseWriter, r *http.Request) {
 	checker := hubconfig.Config.Checker
 	if checker.Check(r) != nil {
 		w.WriteHeader(http.StatusNotFound)
-		w.Write([]byte(fmt.Sprintf("Cloudcore is not ready")))
+		if _, err := w.Write([]byte(fmt.Sprintf("Cloudcore is not ready"))); err != nil {
+			klog.Errorf("failed to write, err: %v", err)
```
Comment (chendave) [inline cloud/pkg/cloudhub/servers/httpserver/server.go:81]: write what? may be "failed to write http response with xxx" or just "write http response error"

File: cloud/pkg/cloudhub/servers/httpserver/server.go [hunk 37]
```diff
@@ -69,18 +69,22 @@ func StartHTTPServer() {
...
+		}
 	} else {
 		w.WriteHeader(http.StatusOK)
-		w.Write([]byte(fmt.Sprintf("Cloudcore is ready")))
+		if _, err := w.Write([]byte(fmt.Sprintf("Cloudcore is ready"))); err != nil {
+			klog.Errorf("failed to write, err: %v", err)
```
Comment (chendave) [inline cloud/pkg/cloudhub/servers/httpserver/server.go:86]: better to be more verbose ,  this is helpful for debugging,  but I am good as is.

File: cloud/pkg/cloudhub/servers/httpserver/server.go [hunk 38]
```diff
@@ -98,13 +102,13 @@ func edgeCoreClientCert(w http.ResponseWriter, r *http.Request) {
 	authorizationHeader := r.Header.Get("authorization")
 	if authorizationHeader == "" {
 		w.WriteHeader(http.StatusUnauthorized)
-		w.Write([]byte(fmt.Sprintf("Invalid authorization token")))
+		_, _ = w.Write([]byte(fmt.Sprintf("Invalid authorization token")))
```
Comment (chendave) [inline cloud/pkg/cloudhub/servers/httpserver/server.go:105]: I don't understand the change like this and other occurrences.

File: cloud/pkg/cloudhub/servers/httpserver/signcerts.go [hunk 39]
```diff
@@ -80,16 +80,18 @@ func GenerateToken() error {
...
 	}
 
 	t := time.NewTicker(time.Hour * 12)
 	go func() {
 		for {
 			<-t.C
 			refreshedCaHashToken := refreshToken()
-			CreateTokenSecret([]byte(refreshedCaHashToken))
+			if err := CreateTokenSecret([]byte(refreshedCaHashToken)); err != nil {
+				klog.Fatalf("failed to craete the ca token for edgecore register, err: %v", err)
```
Comment (chendave) [inline cloud/pkg/cloudhub/servers/httpserver/signcerts.go:93]: s/craete/create

File: cloud/pkg/cloudhub/servers/udsserver/server.go [hunk 40]
```diff
@@ -42,7 +42,10 @@ func StartServer(address string) {
 	})
 
 	klog.Info("start unix domain socket server")
-	uds.StartServer()
+	if err := uds.StartServer(); err != nil {
+		klog.Errorf("failed to start uds server: %v", err)
```
Comment (chendave) [inline cloud/pkg/cloudhub/servers/udsserver/server.go:46]: I am not sure about this, maybe fatal? 

cc @fisherxu  @kevin-wangzefeng  what do you think?

File: cloud/pkg/cloudstream/tunnelserver.go [hunk 41]
```diff
@@ -51,7 +51,7 @@ func newTunnelServer() *TunnelServer {
 			ReadBufferSize:   1024,
 			Error: func(w http.ResponseWriter, r *http.Request, status int, reason error) {
 				w.WriteHeader(status)
-				w.Write([]byte(reason.Error()))
+				_, _ = w.Write([]byte(reason.Error()))
```
Comment (chendave) [inline cloud/pkg/cloudstream/tunnelserver.go:54]: revert this change?

**Comment by chendave** [comment]
btw, lint check is failed as well.

**Comment by daixiang0** [comment]
Thanks, i would fix them.

**Comment by fisherxu** [comment]
Kindly reminder, the review comments without any modification or reply, it should NOT be marked as resolved by the author.

**Comment by daixiang0** [comment]
Sorry for mistake, i would check them again.

File: edgemesh/pkg/proxy/proxy.go [hunk 42]
```diff
@@ -150,13 +150,13 @@ func (p *Proxier) readAndCleanRule() {
 	for scan.Scan() {
 		serverString := scan.Text()
 		if strings.Contains(serverString, "-o") {
-			p.iptables.DeleteRule(utiliptables.TableNAT, utiliptables.ChainOutput, strings.Split(serverString, " ")...)
+			_ = p.iptables.DeleteRule(utiliptables.TableNAT, utiliptables.ChainOutput, strings.Split(serverString, " ")...)
```
Comment (chendave) [inline edgemesh/pkg/proxy/proxy.go:153]: I don't understand those change and other occurrences as well.

Comment (daixiang0) [reply edgemesh/pkg/proxy/proxy.go:153]: This pr is to fix errcheck lint issue, i log err if needed i think. So for some part, just use `_` to fix lint issue.

**Comment by kevin-wangzefeng** [comment]
@daixiang0 Please split config + spcript changes into a separate commit besides fixes.

**Comment by daixiang0** [comment]
@kevin-wangzefeng done.

File: keadm/cmd/keadm/app/keadm.go (hunk 25 above)
Comment (chendave) [reply keadm/cmd/keadm/app/keadm.go:31]: sorry, I am not quite follow you, but consider the default value of `logtostderr` is true I am fine with delete this line. 
question is that change is not related with the errcheck linter, right?

File: tests/stubs/edge/handlerstub/module.go [hunk 43]
```diff
@@ -67,5 +67,8 @@ func (hs *HandlerStub) Start() {
 		klog.Errorf("New upstream controller failed with error: %v", err)
```
Comment (chendave) [inline tests/stubs/edge/handlerstub/module.go:67]: we got no chance that `err` is not nil here, need to revisit how `NewUpstreamController` is defined.

File: tests/stubs/edge/handlerstub/module.go [hunk 44]
```diff
@@ -67,5 +67,8 @@ func (hs *HandlerStub) Start() {
...
 		return
 	}
-	upstream.Start()
+	if err := upstream.Start(); err != nil {
```
Comment (chendave) [inline tests/stubs/edge/handlerstub/module.go:70]: same here, `func (dc *UpstreamController) Start()` return `nil` directly,  there is no chance to reach this code block here.

**Review by chendave — COMMENTED** [review]
leave some questions/comments, otherwise lgtm.

File: tests/stubs/edge/handlerstub/module.go (hunk 43 above)
Comment (daixiang0) [reply tests/stubs/edge/handlerstub/module.go:67]:
```
// Set sets the value of the named command-line flag.
func Set(name, value string) error {
	return CommandLine.Set(name, value)
}
```
pflag.Set would return err.

Now NewUpstreamController always return nil error, but handle it to check is fine for further optimization.

File: tests/stubs/edge/handlerstub/module.go (hunk 44 above)
Comment (daixiang0) [reply tests/stubs/edge/handlerstub/module.go:70]: Same as above.

**Comment by chendave** [comment]
/lgtm

Not perfect, but I don't want to block this.
